source/common/event/dispatcher_impl.cc

```diff
@@ -400,7 +400,7 @@ void DispatcherImpl::pushTrackedObject(const ScopeTrackedObject* object) {
 void DispatcherImpl::popTrackedObject(const ScopeTrackedObject* expected_object) {
   ASSERT(isThreadSafe());
   ASSERT(expected_object != nullptr);
-  RELEASE_ASSERT(!tracked_object_stack_.empty(), "Tracked Object Stack is empty, nothing to pop!");
+  ENVOY_BUG(!tracked_object_stack_.empty(), "Tracked Object Stack is empty, nothing to pop!");
```

Comment:
Looking at the code, this will just blow up in absl equivalent of release_assert in pop_back call below. I think the function can return after ENVOY_BUG if the there is nothing to pop.